Monika After Story/game/definitions.rpy

```diff
@@ -5161,9 +5161,53 @@ init 2 python:
         OUT:
             string prefixed with a/an
         """
-        if ref_str[0] in "aeiou":
-            return "an " + ref_str
-        return "a " + ref_str
+        return ("{0} {1}".format(
+            mas_a_an(
+                ref_str,
+                should_capitalize=ref_str[0].isupper()
+            ),
+            ref_str.lower() if ref_str[0].isupper() and not ref_str.isupper() else ref_str
+        ))
+
+    def mas_a_an(ref_str, should_capitalize):
+        """
+        Takes in a reference string and returns either a/an based on the first letter of the word
+
+        IN:
+            ref_str - string in question to prefix
+
+        OUT:
+            string prefixed with a/an
+        """
+        if ref_str[0] in "aeiouAEIOU":
+            return "An" if should_capitalize else "an"
+        return "A" if should_capitalize else "a"
+
+#EXTRA TEXT TAGS
+init python:
+    def a_an_tag(tag, argument, contents):
+        """
+        Handles a/an mid-string
+
+        NOTE: This should ONLY surround the exact word needing to be prefixed with a/an
+        All text tags should be kept OUTSIDE of the opening and closing tags for this function
+
+        Usage: I bought [player] {a_an}[tempvar]{/a_an}.
+
+        If tempvar was 'item,' the output is: I bought [player] an item.
+        If tempvar was 'coffee,' the output is: I bought [player] a coffee.
+        """
+        list_to_return = list()
+        for kind, _text in contents:
+            #We want to modify only text
+            if kind == renpy.TEXT_TEXT:
+                list_to_return.append((kind, mas_a_an_str(_text)))
+            else:
+                list_to_return.append((kind, _text))
+
+        return list_to_return
```

Comment:
Looks like I can't break it anymore!
What do you think about using `enumerate` to access and modify the only element of the list we need to? Instead of `append`ing every element to a new list?
```suggestion
        for _id, _tuple in enumerate(contents):
            #We want to modify only text
            if _tuple[0] == renpy.TEXT_TEXT:
                contents[_id] = (_tuple[0], mas_a_an_str(_tuple[1]))
                break

        return contents
```